catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -131,179 +121,166 @@ public UpdateResponse version(UpdateResponse updateResponse)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toMap(ContentItem::getId,
+                        ImmutableList::of,
+                        (l, r) -> (List) ImmutableList.builder()
+                                .addAll(l)
+                                .addAll(r)
+                                .build()));
+    }
+
+    private List<ReadStorageRequest> getReadStorageRequests(
+            UpdateStorageResponse updateStorageResponse) {
+        return getReadStorageRequests(updateStorageResponse.getUpdatedContentItems()
+                .stream()
                 .map(ContentItem::getMetacard)
-                .distinct()
+                .collect(Collectors.toList()));
+    }
+
+    private List<ReadStorageRequest> getReadStorageRequests(List<Metacard> metacards) {
+        return metacards.stream()
+                .map(m -> new ReadStorageRequestImpl(m.getResourceURI(),
+                        m.getId(),
+                        new HashMap<>()))
                 .collect(Collectors.toList());
+    }
 
-        // These are being staged for delayed execution, to be ran after the storage/catalog has
-        // successfully committed/updated
-        preStaged.add(() -> {
-            CreateStorageRequestImpl createStorageRequest = new CreateStorageRequestImpl(
-                    versionedContentItems,
-                    new HashMap<>());
-            setSkipFlag(createStorageRequest);
-            CreateStorageResponse createStorageResponseResult = executeAsSystem(() -> {
-                CreateStorageResponse createStorageResponse = storageProvider().create(
-                        createStorageRequest);
-                storageProvider().commit(createStorageRequest);
-                overwriteResourceUris(createStorageResponse);
-                return createStorageResponse;
-            });
-
-            return Optional.ofNullable(createStorageResponseResult)
-                    .map(CreateStorageResponse::getProcessingErrors)
-                    .map(Set::isEmpty)
-                    .orElse(false);
-        });
-
-        preStaged.add(() -> {
-            Map<String, Serializable> props = new HashMap<>();
-            props.put(SKIP_VERSIONING, true);
-            CreateResponse createResponse =
-                    executeAsSystem(() -> catalogProvider().create(new CreateRequestImpl(
-                            versionedMetacards,
-                            props)));
-            return Optional.ofNullable(createResponse)
-                    .map(CreateResponse::getProcessingErrors)
-                    .map(Set::isEmpty)
-                    .orElse(false);
-        });
-
-        return Optional.of(transactionKey);
+    private ReadStorageResponse getStorageItem(ReadStorageRequest r) {
+        try {
+            return storageProvider().read(r);
+        } catch (StorageException e) {
+            LOGGER.info("could not get storage item for metacard (id: {})(uri: {})",
```

Comment:
Added error into logger message. This could happen multiple times so not sure we want to log it at Error.